### DISCUSSION TITLE
MissionControl 0.9.4

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#714] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.9.4/total">
<a href="https://discord.gg/gegfNZ5Ucz"><img alt="Discord Server" src="https://discordapp.com/api/guilds/905069757361971303/widget.png?style=shield"/></a>
</p>

This is the twenty-first official release of Mission Control.

This is a hotfix release to address the freezing issue some users reported when pairing controllers on 16.1.0

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl#readme) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

_PS: there is some ongoing testing of preliminary Bluetooth LE support happening in my discord server (see #testing). If you own any of the controllers listed [here](https://github.com/ndeadly/MissionControl/issues?q=is%3Aissue+is%3Aopen+label%3A%22BLE+controller%22) or any LE controllers that are not on the list, we'd like to hear from you! (though I don't need any additional input on Xbox and Stadia controllers for now). Please note that these are alpha builds for the purpose of gathering controller information and furthering my understanding of how BLE works on the console. They are **_NOT_** intended for general use. There _will_ be bugs and issues. There are no ETAs. I will not be providing any support - if you can't follow the usage instructions, then this is not for you._

### Changelog:
- Fixed a bug in the exefs patches for 16.1.0 that caused the system to get stuck in an infinite loop and freeze when pairing new controllers not already in the bluetooth pairing database
- Check vendor/product ID in addition to name when identifying official Switch controllers

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.9.4'>MissionControl 0.9.4</a>.</em>

**Comment by a0930530** [comment]
Wiiu pro can be used, 

ps4 controller can not be used，ps+share unable to pair  not available ps4 controlle

**Reply by a0930530** [reply]
0.92+16.0.3+1.5.3 can use PS4，The latest version cannot be used

**Comment by Ntemis** [comment]
@a0930530 do you know what is your ps4 controller model?

**Reply by a0930530** [reply]
cuh-zct1t，AND The font of the second controller disappears

**Reply by a0930530** [reply]
CUH-ZCT2 ，Another controller font disappears

**Reply by a0930530** [reply]
CUH-ZCT2?  The font is gone

**Reply by Ntemis** [reply]
you mean it connects and disconnects?

**Reply by a0930530** [reply]
Unable to connect to controller, no response

**Reply by a0930530** [reply]
0.92+16.0.3+1.5.3 can use PS4，The latest version cannot be used

**Reply by ndeadly** [reply]
Try clearing the bluetooth pairing database and resetting your controller. PS4 controllers are definitely working.

**Reply by a0930530** [reply]
The latest version cannot be used PS4 controller，0.94+16.1.0+1.5.5

**Reply by ndeadly** [reply]
It can. I'm using one right now. Why do you insist on giving as little information as possible?

**Reply by a0930530** [reply]
Only certain models of your PS4 can be paired

**Reply by ndeadly** [reply]
Please stop commenting if you are not going to provide any useful information. Simply stating that something doesn't work contributes nothing toward finding a solution. All official Sony controllers are known to work.

**Reply by zituzz** [reply]
@ndeadly Hello I just install the latest version of the program in my switch and the ps4 controller not work for me. I test it on 16.1.3

**Reply by ndeadly** [reply]
@zituzZ make sure you've read https://github.com/ndeadly/MissionControl#usage, https://github.com/ndeadly/MissionControl#pairing-controllers and https://github.com/ndeadly/MissionControl#frequently-asked-questions. The FAQ outlines some general troubleshooting steps. Also, with PS controllers sometimes they need to be reset after having been connected to other devices. There's a little pinhole button on the underside.

**Comment by ZaydelSenpai** [comment]
Also stock Dualsense not working for me, led just blink and then dissapear.
CFI-ZCT1W

**Reply by ndeadly** [reply]
Make sure the controller has sufficient battery, and try resetting it via the little pinhole button underneath. If that doesn't work, refer to the [FAQ ](https://github.com/ndeadly/MissionControl#frequently-asked-questions) section titled `I installed Mission Control but it doesn't do anything. Help!` for troubleshooting tips.

**Reply by ZaydelSenpai** [reply]
I restarted the console, controller settings, unpair, sync again, charged the controller, and sysmodules also showing MissionControl it’s working, also paired a Xbox one controller but my dualsense still not pairing 🤔

**Reply by ndeadly** [reply]
> try resetting it via the little pinhole button underneath

Did you do this? You could also try deleting `/config/MissionControl/controllers` from your sd card.

> sysmodules also showing MissionControl

Assuming you mean the Tesla sysmodules overlay? This doesn't know the module is running, FYI. It only checks whether the flag file is present to tell atmosphere it _should_ be loaded. It doesn't account for other issues (eg. archive bits) preventing the module from being loaded. That said, if you paired an Xbox controller then it's running.

**Comment by Edisjc** [comment]
Hellow ndeadly
I have a system crash promlem when use MissionControl 0.9.4 , can you help? or tell me how to solve
system info : FW 16.1.0
controller : Sony DualShock4 (Playstation 4) Controller gen1/gen2
Issue : when press SHARE+PS from controller, system will crash when controller connect
pic : https://imgur.com/I1TXqs5

**Reply by Edisjc** [reply]
Response, I remove all of paired BT controller information before FW update to 16.1.0, then re-pair, issue fix.
Thanks for missioncontrol app and designer.

**Reply by ndeadly** [reply]
Would you mind posting the error report referenced in the screenshot you linked? Even though you were able to resolve your problem, this may be an issue that I can fix so that others don't experience such crashes.

**Comment by Jaenani** [comment]
I think I found a bug, I apologize if this is known about. 

When using the NSO N64 controller in atomosphere with mission control enabled c-right becomes a key modifier and pressing d-pad up will take a screenshot and d-pad down will be a shortcut for home. (not many games use the d-pad but wanted to play Ready2Rumble boxing with it and noticed it)

At first I thought this was maybe intended because I didn't recall a screenshot button on the controller but low and behold right on the top it was there. Took me a FAT minute to realize mission control was the cause as I've had mission control installed a few weeks before I got the controller in. Anywho, disabling mission control autostart and rebooting remedies this issue so not a huge deal just thought I'd report it in case it wasn't known about. Cheers!

**Reply by ndeadly** [reply]
Heh, I'm surprised nobody has reported this one sooner. I didn't know about it, but it makes sense why this happens.

Mission control implements button combos for home and capture buttons using the minus and dpad down/up buttons. These are intended for controllers that don't have enough buttons to map those keys, but are applied to all controllers. The n64 controller is unique in that it doesn't use the same button layout to the other official controllers in the input data it sends. The combo code assumes the original data format, so for n64 controllers it will not necessarily be checking the inputs it thinks it is.

Could you please open this as official issue report so I don't forget about it? I'm in the process of refactoring some of the controller code and I will take this into account and fix it in the process.
